eth/peerset.go

```diff
@@ -402,10 +404,27 @@ func (ps *peerSet) waitBscExtension(peer *eth.Peer) (*bsc.Peer, error) {
 		return peer, nil
 
 	case <-time.After(extensionWaitTimeout):
-		ps.lock.Lock()
-		delete(ps.bscWait, id)
-		ps.lock.Unlock()
-		return nil, errPeerWaitTimeout
+		// once timeout is reached, bsc protocol won't be enabled on this peer.
+		log.Warn("waitBscExtension", "peer", id, "bsc protocol timeout, it won't be enabled.")
+		// could be deadlock, so we use TryLock to avoid it.
```

Comment:
can delete 409～414？

Reply:
yes, can be deleted, but it will cause an extra tryWaitTimeout, so not prefer to delete it.